### PR TITLE
fix: Failing tests

### PR DESCRIPTION
**Pull request by hariombalhara**
E2E Tests Status - https://github.com/calcom/cal.com/actions/runs/7448429724

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #13091** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/13091?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/13091?utm_source=stack-comment).

**Review by PeerRich — DISMISSED** [review]

File: apps/web/playwright/fixtures/users.ts [hunk 1]
```diff
@@ -751,7 +751,7 @@ export async function makePaymentUsingStripe(page: Page) {
   const stripeFrame = stripeElement.frameLocator("iframe").first();
   await stripeFrame.locator('[name="number"]').fill("4242 4242 4242 4242");
   const now = new Date();
-  await stripeFrame.locator('[name="expiry"]').fill(`${now.getMonth()} / ${now.getFullYear() + 1}`);
+  await stripeFrame.locator('[name="expiry"]').fill(`${now.getMonth() + 1} / ${now.getFullYear() + 1}`);
```
Comment (hariombalhara) [inline apps/web/playwright/fixtures/users.ts:754]: January was translating to 0 which is invalid.

File: packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx [hunk 2]
```diff
@@ -553,11 +553,18 @@ export default function RouteBuilder({
...
       appUrl={appUrl}
-      Page={({ hookForm, form }) => (
-        <div className="route-config">
-          <Routes hookForm={hookForm} appUrl={appUrl} form={form} />
-        </div>
-      )}
+      Page={({ hookForm, form }) => {
+        // If hookForm hasn't been initialized, don't render anything
+        // This is important here because some states get initialized which aren't reset when the hookForm is reset with the form values and they don't get the updated values
+        if (!hookForm.getValues().id) {
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx:559]: The code was flaky earlier, but no idea why it's consistently failing now but it's the right fix.

**Review by emrysal — APPROVED** [review]
Good call on adding the TODO, let's merge.

**Pull request merged**